x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+//! A basic NFT contract.
+//! The program serves as a non-fungible token with the ability to mint and burn.
+//! Only supports whole units with no decimal places.
+//!
+//! The NFT must support the common NFT metadata format.
+//! This includes the name, symbol, and URI of the NFT.
+use wasmlanche_sdk::{
+    memory::{Memory, Pointer},
+    program::Program,
+    public, state_keys,
+    types::Address,
+};
+
+/// The total supply of the NFT.
+/// # Safety
+/// In this program only one NFT of each type is intended to be minted.
+const TOTAL_SUPPLY: i64 = 1;
```

Comment:
Why store `TotalSupply` in state if it is a const? I think it would be better (upon review) to remove this const and just make it configurable below (the way you are forcing a 1 of 1 here is confusing).

If we want to keep as a 1 of 1, you should remove `TotalSupply` from state.

Reply:
It's there for illustration purposes really. We can remove it entirely.